### PR TITLE
modify date time to ease parsing and serailizing

### PR DESCRIPTION
**Pull request by clementpeihengtan**

### CODE REVIEW

File: Cogs.Publishers/JsonSimpleConverter.cs [hunk 1]
```diff
@@ -31,7 +31,7 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 obj.Add(new JProperty("dateTime", 
                     new JObject(
                         new JProperty("type", "string"), 
-                        new JProperty("format", "date-time"))));
+                        new JProperty("pattern", @"^[0-9][0-9][0-9][0-9]-(0?[1-9]|1[0-2])-(0?[1-9]|[12][0-9]|3[01])([A-Z])(00|0[0-9]|1[0-9]|2[0-3]):([0-9]|[0-5][0-9]):([0-9]|[0-5][0-9])(\+)(00|0[0-9]|1[0-9]|2[0-3]):([0-9]|[0-5][0-9]):([0-9]|[0-5][0-9])$"))));
```
Comment (DanSmith) [inline Cogs.Publishers/JsonSimpleConverter.cs]: Why does the built in date-time format not work?

Comment (clementpeihengtan) [inline Cogs.Publishers/JsonSimpleConverter.cs]: The build in date-time format, only accept 
"2012-04-23T18:25:43.511Z"     
"2012-04-21T18:25:43-05:00"
But not
"2012-04-21T18:25:43-05:00:00"
Since the offset comes in the form of  XX:XX:XX, it is hard to parse these value and convert it back and forth.

Comment (kevinchristianson) [inline Cogs.Publishers/JsonSimpleConverter.cs]: I tried serializing datetime in the 0000-00-00T00:00:00-00:00 format but since Datetime in c# has an extra :00 at the end of the timespan the newtonsoft JObject.Parse method would incorrectly parse the data stored

Comment (DanSmith) [inline Cogs.Publishers/JsonSimpleConverter.cs]: We can use the string format "yyyy-MM-dd\THH:mm:ss.FFFFFFFK" in c# to read in and write out the iso 8601 time format
